tests/conftest.py

```diff
@@ -0,0 +1,93 @@
+# -*- coding: utf-8 -*-
+from datetime import datetime
+
+import pytest
+from dateutil import tz
+
+from arrow import arrow, factory, formatter, locales, parser
+
+
+@pytest.fixture(scope="class")
+def utcnow_fixture(request):
+    request.cls.arrow = arrow.Arrow.utcnow()
+
+
+@pytest.fixture(scope="class")
+def time2013_01_01_fixture(request):
+    request.cls.now = arrow.Arrow.utcnow()
+    request.cls.arrow = arrow.Arrow(2013, 1, 1)
+    request.cls.datetime = datetime(2013, 1, 1)
+
+
+@pytest.fixture(scope="class")
+def time2013_02_03_fixture(request):
+    request.cls.arrow = arrow.Arrow(2013, 2, 3, 12, 30, 45, 1)
+
+
+@pytest.fixture(scope="class")
+def time2013_02_15_fixture(request):
+    request.cls.datetime = datetime(2013, 2, 15, 3, 41, 22, 8923)
+    request.cls.arrow = arrow.Arrow.fromdatetime(request.cls.datetime)
+
+
+@pytest.fixture(scope="class")
+def timedst_fixture(request):
+    request.cls.before_1 = arrow.Arrow(
+        2016, 11, 6, 3, 59, tzinfo=tz.gettz("America/New_York")
+    )
+    request.cls.before_2 = arrow.Arrow(2016, 11, 6, tzinfo=tz.gettz("America/New_York"))
+    request.cls.after_1 = arrow.Arrow(
+        2016, 11, 6, 4, tzinfo=tz.gettz("America/New_York")
+    )
+    request.cls.after_2 = arrow.Arrow(
+        2016, 11, 6, 23, 59, tzinfo=tz.gettz("America/New_York")
+    )
+    request.cls.before_3 = arrow.Arrow(
+        2018, 11, 4, 3, 59, tzinfo=tz.gettz("America/New_York")
+    )
+    request.cls.before_4 = arrow.Arrow(2018, 11, 4, tzinfo=tz.gettz("America/New_York"))
+    request.cls.after_3 = arrow.Arrow(
+        2018, 11, 4, 4, tzinfo=tz.gettz("America/New_York")
+    )
+    request.cls.after_4 = arrow.Arrow(
+        2018, 11, 4, 23, 59, tzinfo=tz.gettz("America/New_York")
+    )
+
+
+@pytest.fixture(scope="class")
+def formatting_fixture(request):
+    request.cls.formatter = formatter.DateTimeFormatter()
+
+
+@pytest.fixture(scope="class")
+def locales_fixture(request):
+    request.cls.locales = locales._locales
+
+
+@pytest.fixture(scope="class")
+def lang_locale_fixture(request):
+    name = request.cls.__name__[4:]
```

Comment:
Hah this is definitely a hack. Let's put a comment to prevent confusion and explain the thought process.

Reply:
added a comment + example.